bin/detect

```diff
@@ -4,10 +4,19 @@ set -eo pipefail
 
 bp_dir=$(cd "$(dirname "$0")"/..; pwd)
 build_dir=$(pwd)
+build_plan="$2"
 
 # shellcheck source=/dev/null
 source "$bp_dir/lib/detect.sh"
 
 if ! detect_package_json "$build_dir" ; then
   exit 100
+else
+  cat << EOF > "$build_plan"
+  [[provides]]
+  name = "node_modules"
+
+  [[requires]]
+  name = "node"
+EOF
```

Comment:
Is this something that we want to add as a `buildplan.toml` file to the buildpack so that we don't have to write it at build time?

Reply:
I don't think you can. The "$2" is the buildplan that is built based on the `detect` phase.

Reply:
Gotcha 👍 in that case, do you mind if we move this to a method? It will make it easier to unit test when we add them. You can create a `lib/detect.sh` file and then just import the method back here like we do with `lib/build.sh` and `bin/build`:

https://github.com/heroku/nodejs-npm-buildpack/blob/master/bin/build#L15
https://github.com/heroku/nodejs-npm-buildpack/blob/master/lib/build.sh#L31-L40

Reply:
Also do we need to corollate the name with the layer name? If so, we will want the name on L20 to be `nodejs`

Reply:
I went ahead and moved it to a function and added a test. The name of what we provide needs to match the name given by downstream buildpacks. In this case our only real world example is riff and they require `node` and `node_modules` for the `node-function-buildpack`. We are discussing maybe settling on something like `io.buildpacks.node` or something that is more of a contract for language buildpacks so that buildpack owners can, in the future, be compatible with both CF and HK buildpacks as long as we agree on a few things and share the `provides` name.